### ISSUE TITLE
Installing in v.0.6: Build Error: 'src' and 'dst' refer to the same file/dir.This is not supported.

### ISSUE THREAD

**Issue by TomCab** [JuliaMath/SpecialFunctions.jl#53]
# Description of the problem

While attempting to do a fresh install of SpecialFunctions in Julia v. 0.6.0 on Fedora 26 Linux (x86_64), the package fails to compile. Following error is reported:

```
julia> Pkg.add("SpecialFunctions")
INFO: Cloning cache of SpecialFunctions from https://github.com/JuliaMath/SpecialFunctions.jl.git
INFO: Installing SpecialFunctions v0.3.2                                                                                                    
INFO: Building SpecialFunctions                                                                                                             
INFO: No precompiled binaries found for your system. Building from scratch...                                                               
========================================================[ ERROR: SpecialFunctions ]=========================================================

LoadError: LoadError: ArgumentError: 'src' and 'dst' refer to the same file/dir.This is not supported.
  `src` refers to: /usr/lib64/libopenlibm.so.2                                                                                              
  `dst` refers to: /usr/lib64/libopenlibm.so.2                                                                                              
                                                                                                                                            
while loading /home/<my-user-name>/.julia/v0.6/SpecialFunctions/deps/scratch.jl, in expression starting on line 8
while loading /home/<my-user-name>/.julia/v0.6/SpecialFunctions/deps/build.jl, in expression starting on line 40

============================================================================================================================================

==============================================================[ BUILD ERRORS ]==============================================================

WARNING: SpecialFunctions had build errors.
                                                                                                                                            
 - packages with build errors remain installed in /home/<my-user-name>/.julia/v0.6                                                                     
 - build the package(s) and all dependencies with `Pkg.build("SpecialFunctions")`                                                           
 - build a single package by running its `deps/build.jl` script                                                                             

============================================================================================================================================

```
I removed the package through
```
julia> Pkg.rm("SpecialFunctions")
```
then I quit the Julia REPL, deleted the package from Julia's cache in my home directory, restarted the Julia REPL and tried reinstalling , with the same result.

I found in my system two different versions of ```libopenlibm.so```:
- ```/usr/lib64/libopenlibm.so.2```
- ```/usr/lib64/libopenlibm.so.2.3```

but I don't know whether this has anything to do with the build problems at all.

# System information
```
julia> versioninfo()
Julia Version 0.6.0
Commit 903644385b* (2017-06-19 13:05 UTC)
Platform Info:
  OS: Linux (x86_64-redhat-linux)
  CPU: AMD FX(tm)-6300 Six-Core Processor
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH Piledriver)
  LAPACK: libopenblasp.so.0
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, bdver1)
```

**Comment by ararslan** [comment]
Thanks for the thorough report. I have a fix in mind that I'll push ASAP. In the meantime you can do `Pkg.pin("SpecialFunctions", v"0.3.1")` if you need this working right away.

**Comment by ararslan** [comment]
This should be fixed on master and in v0.3.3. Please let me know if you run into it again.

**Comment by TomCab** [comment]
Great, ```v0.3.3``` builds without any problems.

Thanks.
